From: Daniel Baker <dbaker@mozilla.com>
Date: Fri, 15 Mar 2024 17:24:40 -0600
Subject: (tmp-cherry-pick) Revert "Fix 'Image will be cropped if
 WindowCapturerWinGdi used'" (634cb403e6)

This reverts commit 844225a76a98aa3be5aca09c19ab72a5e7b6c38a.

Reason for revert: potential nullptr dereference

Original change's description:
> Fix 'Image will be cropped if WindowCapturerWinGdi used'
>
> Bug: webrtc:15719
> Change-Id: I7daf8ee5b90fbe9f1246f1d99211ffa0d8a19f73
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/330780
> Reviewed-by: Alexander Cooper <alcooper@chromium.org>
> Commit-Queue: Alexander Cooper <alcooper@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#41503}

Bug: webrtc:15719
Change-Id: Ib38e1345c4c590b6a71bbea476a9d780a2f5e800
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/334200
Owners-Override: Jeremy Leconte <jleconte@google.com>
Commit-Queue: Jeremy Leconte <jleconte@google.com>
Reviewed-by: Manashi Sarkar <manashi@google.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#41509}
---
 AUTHORS                                             | 1 -
 modules/desktop_capture/win/window_capture_utils.cc | 7 -------
 2 files changed, 8 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index 968c896856..6e922401d0 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -120,7 +120,6 @@ Silviu Caragea <silviu.cpp@gmail.com>
 Stefan Gula <steweg@gmail.com>
 Stephan Hartmann <stha09@googlemail.com>
 Steve Reid <sreid@sea-to-sky.net>
-Tao chen <meemetao@gmail.com>
 Takaaki Suzuki <takaakisuzuki.14@gmail.com>
 Tarun Chawla <trnkumarchawla@gmail.com>
 Todd Wong <todd.wong.ndq@gmail.com>
diff --git a/modules/desktop_capture/win/window_capture_utils.cc b/modules/desktop_capture/win/window_capture_utils.cc
index 13f498f332..ccfef49bc5 100644
--- a/modules/desktop_capture/win/window_capture_utils.cc
+++ b/modules/desktop_capture/win/window_capture_utils.cc
@@ -183,13 +183,6 @@ bool GetCroppedWindowRect(HWND window,
       is_maximized) {
     // Only apply this cropping to windows with a resize border (otherwise,
     // it'd clip the edges of captured pop-up windows without this border).
-    RECT rect;
-    DwmGetWindowAttribute(window, DWMWA_EXTENDED_FRAME_BOUNDS, &rect,
-                          sizeof(RECT));
-    // it's means that the window edge is not transparent
-    if (rect.left == original_rect->left()) {
-      return true;
-    }
     LONG style = GetWindowLong(window, GWL_STYLE);
     if (style & WS_THICKFRAME || style & DS_MODALFRAME) {
       int width = GetSystemMetrics(SM_CXSIZEFRAME);
